### PR TITLE
[3279] Restore stereotypes retrieval and document creation

### PR DESCRIPTION
**Pull request by sbegaudeau**
- [3279] Restore stereotypes retrieval and document creation
- [cleanup] Fix invalid copyright headers

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioStereotypeHandler.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    private Optional<DocumentDTO> createDomainDocument(IEMFEditingContext editingContext, String name) {
+        var documentId = UUID.randomUUID();
+        var resource = new JSONResourceFactory().createResourceFromPath(documentId.toString());
+        resource.eAdapters().add(new ResourceMetadataAdapter(name));
+
+        editingContext.getDomain().getResourceSet().getResources().add(resource);
+
+        Domain domain = DomainFactory.eINSTANCE.createDomain();
+        domain.setName("SampleDomain");
+        domain.setName(this.domainNameProvider.getSampleDomainName());
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioStereotypeHandler.java:78]: We set the name two times

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/documents/DocumentControllerIntegrationTests.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright (c) 2024, 2024 Obeo.
+ * This program and the accompanying materials
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/documents/DocumentControllerIntegrationTests.java:3]: Wrong dates

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/FormControllerIntegrationTests.java [hunk 3]
```diff
@@ -34,6 +34,7 @@
 import org.eclipse.sirius.components.forms.tests.graphql.EditSelectMutationRunner;
 import org.eclipse.sirius.components.forms.tests.navigation.FormNavigator;
 import org.eclipse.sirius.web.AbstractIntegrationTests;
+import org.eclipse.sirius.web.data.StudioIdentifiers;
 import org.eclipse.sirius.web.data.TestIdentifiers;
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/FormControllerIntegrationTests.java:38]: Wrong dates in the copyright

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/RelatedElementsViewControllerIntegrationTests.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
 import org.eclipse.sirius.components.forms.tests.graphql.RelatedElementsEventSubscriptionRunner;
 import org.eclipse.sirius.components.forms.tests.navigation.FormNavigator;
 import org.eclipse.sirius.web.AbstractIntegrationTests;
-import org.eclipse.sirius.web.data.TestIdentifiers;
+import org.eclipse.sirius.web.data.StudioIdentifiers;
 import org.eclipse.sirius.web.services.api.IGivenInitialServerState;
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/RelatedElementsViewControllerIntegrationTests.java:32]: Wrong dates in the copyright

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/data/TestIdentifiers.java [hunk 5]
```diff
@@ -22,17 +22,14 @@
 public final class TestIdentifiers {
 
     public static final UUID INVALID_PROJECT = UUID.fromString("a2bed581-9661-41bf-9217-2870a9dce67c");
-    public static final UUID EMPTY_STUDIO_PROJECT = UUID.fromString("250cabc0-a211-438c-8015-2d2aa136eb81");
     public static final UUID SAMPLE_STUDIO_PROJECT = UUID.fromString("01234836-0902-418a-900a-4c0afd20323e");
-    public static final UUID FORM_DESCRIPTION_OBJECT = UUID.fromString("ed20cb85-a58a-47ad-bc0d-749ec8b2ea03");
     public static final UUID UML_SAMPLE_PROJECT = UUID.fromString("7ba7bda7-13b9-422a-838b-e45a3597e952");
```
Comment (frouene) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/data/TestIdentifiers.java:26]: Wrong dates in the copyright

**Review by frouene — CHANGES_REQUESTED** [review]

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/data/TestIdentifiers.java (hunk 5 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/data/TestIdentifiers.java:26]: Done

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/RelatedElementsViewControllerIntegrationTests.java (hunk 4 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/RelatedElementsViewControllerIntegrationTests.java:32]: Done

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/FormControllerIntegrationTests.java (hunk 3 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/forms/FormControllerIntegrationTests.java:38]: Done

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/documents/DocumentControllerIntegrationTests.java (hunk 2 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/documents/DocumentControllerIntegrationTests.java:3]: Done

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioStereotypeHandler.java (hunk 1 above)
Comment (sbegaudeau) [reply packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/StudioStereotypeHandler.java:78]: Done

**Review by frouene — APPROVED** [review]

**Pull request merged**